voltixApp/VoltixApp/View Models/KeysignViewModel.swift

```diff
@@ -1,279 +1,286 @@
-        case HelperError.runtimeError(let errDetail):
-            self.logger.error("Failed to get signed transaction,error:\(errDetail)")
-        default:
-            self.logger.error("Failed to get signed transaction,error:\(err.localizedDescription)")
-        }
-    }
-    
-    func handleBitcoinTransactionError(err: UTXOTransactionError) {
-        switch err {
-        case .invalidURL:
-            print("Invalid URL.")
-        case .httpError(let statusCode):
-            print("HTTP Error with status code: \(statusCode).")
-        case .apiError(let message):
-            print("API Error: \(message)")
-        case .unexpectedResponse:
-            print("Unexpected response from the server.")
-        case .unknown(let unknownError):
-            print("An unknown error occurred: \(unknownError.localizedDescription)")
-        }
-    }
+	func tssKeysign(service: TssServiceImpl, req: TssKeysignRequest, keysignType: KeyType) async throws -> TssKeysignResponse {
+		let t = Task.detached(priority: .high) {
+			switch keysignType {
+				case .ECDSA:
+					return try service.keysignECDSA(req)
+				case .EdDSA:
+					return try service.keysignEdDSA(req)
+			}
+		}
+		return try await t.value
+	}
+	
+	func broadcastTransaction() async {
+		if let keysignPayload {
+			if keysignPayload.swapPayload != nil {
+				let swaps = THORChainSwaps(vaultHexPublicKey: vault.pubKeyECDSA, vaultHexChainCode: self.vault.hexChainCode)
+				let result = swaps.getSignedTransaction(keysignPayload: keysignPayload, signatures: self.signatures)
+				switch result {
+					case .success(let tx):
+						print(tx)
+					case .failure(let err):
+						print(err.localizedDescription)
+				}
+				return
+			}
+			switch keysignPayload.coin.chain.chainType {
+				case .UTXO:
+					
+					let chainName = keysignPayload.coin.chain.name.lowercased()
+					
+					guard let coinType = CoinType.from(string: chainName) else {
+						print("Coin type not found on Wallet Core")
+						return
+					}
+					
+					let utxoHelper = UTXOChainsHelper(coin: coinType, vaultHexPublicKey: vault.pubKeyECDSA, vaultHexChainCode: self.vault.hexChainCode)
+					
+					let result = utxoHelper.getSignedTransaction(keysignPayload: keysignPayload, signatures: self.signatures)
+					
+					switch result {
+						case .success(let tx):
+							print("Broadcasting UTXO transaction from \(chainName): \(tx)")
+							UTXOTransactionsService.broadcastTransaction(chain: chainName, signedTransaction: tx) { result in
+								switch result {
+									case .success(let transactionHash):
+										self.txid = transactionHash
+										print("Transaction successfully broadcasted. Hash: \(transactionHash)")
+									case .failure(let error):
+										print("Error broadcasting transaction: \(error.localizedDescription)")
+								}
+							}
+						case .failure(let err):
+							self.handleHelperError(err: err)
+					}
+					
+				case .EVM:
+					// ETH
+					if !keysignPayload.coin.isToken {
```

Comment:
I will say  `isToken` is a bit weird , ETH and ERC20 they are both tokens ,  maybe we could come up a better name

for example: `IsNativeToken` ?

Reply:
Renamed